tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/GitStore.cs

```diff
@@ -219,13 +219,24 @@ public HttpException GenerateInvokeException(CommandResult result)
             return new HttpException(HttpStatusCode.InternalServerError, message);
         }
 
+        private void SetSafeDirectory(GitAssetsConfiguration config)
+        {
+            // Workaround for git directory ownership checks that may fail when running in a container as a different user.
+            if ("true" == Environment.GetEnvironmentVariable("TEST_PROXY_CONTAINER"))
+            {
+                GitHandler.Run($"config --global --add safe.directory {config.AssetsRepoLocation}", config);
+            }
+        }
+
         /// <summary>
         /// Checks an asset repository for pending changes. Equivalent of "git status --porcelain".
         /// </summary>
         /// <param name="config"></param>
         /// <returns></returns>
         public string[] DetectPendingChanges(GitAssetsConfiguration config)
         {
+            SetSafeDirectory(config);
```

Comment:
Is this going to be something we're likely to add to more and more places in the code as we discover safe directory issues? If so, would it be better to move `SetSafeDirectory` into `GitHandler.Run`?

Reply:
As long as we run it _at least once_ before we run a git command, yes. I didn't place it directly in githandler.run because I don't like the amount of duplicated invocation of this exact statement that we will be adding.

Reply:
Yeah I agree I don't like basically doubling all our git calls in container mode.